src/app/day-calculator-modal/day-calculator-modal.component.ts

```diff
@@ -174,126 +180,152 @@ export class DayCalculatorModalComponent implements OnInit {
-    // Given two of Start Date, End Date, and Number of Days, calculate the third field.
-
-    if (this.startDate && this.endDate) {
+        if (startDateMoment >= endDateMoment) {
+          // Show error if start date is later than end date
+          throw new Error('The start date must come before the end date.');
+        }
 
-      if (startDateMoment >= endDateMoment) {
-        // Show error if start date is later than end date
-        this.errorAlert = 'The start date must come before the end date.';
-        return;
-      }
+        // Include the start date in the calculation.
+        calcRes.numDays = regular ? 1 : 0;
 
-      // Include the start date in the calculation.
-      this.numDays = regular ? 1 : 0;
+        // Count the number of days between start and end dates
+        let date = startDateMoment;
 
-      // Count the number of days between start and end dates
-      let date = startDateMoment;
+        // Look at every date between
+        while (date < endDateMoment) {
+          date.add(1, 'd');
 
-      // Look at every date between
-      while (date < endDateMoment) {
-        date.add(1, 'd');
+          // Factor in a suspension
+          if (suspended && suspendDate && date >= suspendDateMoment) {
+            // Stop if the suspension goes past the end date
+            if (!resumeDate || endDateMoment < resumeDateMoment) {
+              return calcRes;
+            }
 
-        // Factor in a suspension
-        if (suspended && this.suspendDate && date >= suspendDateMoment) {
-          // Stop if the suspension goes past the end date
-          if (!this.resumeDate || endDateMoment < resumeDateMoment) { return; }
+            // Don't count days in the suspension range
+            if (date < resumeDateMoment) { continue; }
+          }
 
-          // Don't count days in the suspension range
-          if (date < resumeDateMoment) { continue; }
+          // If we've made it this far, count the day
+          calcRes.numDays++;
         }
-
-        // If we've made it this far, count the day
-        this.numDays++;
-      }
-    } else if (this.startDate && this.numDays) {
-      // Find the end date from the start date and number of days
-      endDateMoment = moment(this.startDate);
-
-      // Include the start date in the calculation.
-      numDays = regular ? 1 : 0;
-
-      // Start counting the days
-      while (numDays < this.numDays) {
-        endDateMoment.add(1, 'd');
-
-        // Factor in a suspension
-        if (suspended && this.suspendDate && endDateMoment >= suspendDateMoment) {
-          if (!this.resumeDate) {
-            // Can't find an end date if there is no resume date
-            this.endDate = '';
-            return;
+      } else if (startDate && numDays) {
+        // Find the end date from the start date and number of days
+        endDateMoment = moment(startDate);
+
+        // Include the start date in the calculation.
+        numDays = regular ? 1 : 0;
```

Comment:
should this be 
`calcRes.numDays = regular ? 1 : 0;`

Reply:
originally was 
numDays = regular ? 1 : 0;